.changeset/slimy-chicken-mix.md

```diff
@@ -0,0 +1,6 @@
+---
+'@firebase/firestore': patch
+'firebase': patch
+---
+
+Reverted a change to use UTF-8 encoding in string comparisons which caused a performance issue.
```

Comment:
Another drive-by comment: Append `Fixes #NNNN` to allow interested parties to click through to the github issue and understand more context.